include/class.file.php

```diff
@@ -358,6 +356,10 @@ static function uploadLogo($file, &$error, $aspect_ratio=2) {
     }
 
     static function create(&$file, $ft='T', $deduplicate=true) {
+        // Crap-out if file contents do not match expected file type
+        if (strpos('image/', $file['type']) !== 0 && !exif_imagetype($file['tmp_name']))
```

Comment:
This will crap out when an image is emailed - liked I mentioned in this thread, the logic  NEEDS to go to FileUploadField. I'll make a PR once I have it tested.

Reply:
My mistake, I apologize, I don't know how I missed that part of the comment. Must've had a lot going on that day I guess.